solrmonitor/zkdispatcher.go

```diff
@@ -91,6 +116,11 @@ func (d *ZkDispatcher) Watch(watcher <-chan zk.Event, handler ZkEventHandler) er
 	return nil
 }
 
+// Watch a new ZK event using the given function to handle the event.
+func (d *ZkDispatcher) Watch(watcher <-chan zk.Event, handler ZkEventHandler) error {
```

Comment:
Retained for backwards compatibility of the API. But functions can't be used as map keys, so they can't be used to adequately enforce serial delivery of events (in the event one tried t use the same function as the handler for multiple watches) since we can't actually tell if two invocations of this method are providing the same function or not. See new `WatchEvent` method.